### PR TITLE
side menu documentation

### PR DESCRIPTION
**Pull request by victorscania**
**Describe pull-request**  
Just so you can read the updated documentation for the side menu

**Solving issue**  
_Add which issue this pull-request solves by adding # plus the number of the issue (for example #123)_
Fixes: [AB#1692](https://dev.azure.com/scaniadigitaldesignsystem/3d9eb9c6-0045-473d-bc1d-e842dd779200/_workitems/edit/1692)

**How to test**  
Read the documentation on the side-menu from the generated link

### CODE REVIEW

**Comment by theJohnnyMe** [comment]
I think beta tag from Code 2.0 can be removed: 
beta_tag_1

Do we want to keep the Beta tag on the current one and for how long? I have no opinion there.

**Comment by theJohnnyMe** [comment]
I think a better tab name would be "Code => 2.1.0" as that is our last release of @scania/components that has old code. I think our next release would be 3.0.0 due to Breaking changes.

code_version

**Comment by theJohnnyMe** [comment]
You can get rid off body tag and sdds-theme one in code examples.

example_code

**Comment by theJohnnyMe** [comment]
@victorscania I added a couple of things up here that can be changed and improved. Overall, great work! ;)

**Comment by areblk** [comment]
> I think a better tab name would be "Code => 2.1.0" as that is our last release of @scania/components that has old code. I think our next release would be 3.0.0 due to Breaking changes.
> 
> code_version

I'd suggest to name it "Code 2.*" because it's a format that can be used for all breaking changes documentation. Url can be /code-2

**Comment by areblk** [comment]
1. For convinience and aesthetics, url to latest code doc should be '/components/sidemenu/code'
2. No reason to repeat information that's in usage tab. Introduction section can be removed. If the headline is part of convention, rename the 'Basic structure' to 'Introduction'.
3. Remove "In Beta - Work in progress." I wouldn't say this is more Beta than any other of our components. 
4. Use same language as in usage tab. 
- "Side menu items" --> "Page links"
- "Side menu item with sub page options" --> "Sub-menus with page links"
- "Toggle the display of the sup page items" --> "Toggle the display of page links in sub-menu"
- "Toggle a sup page options active state" --> "Toggle a sub-menu or page link active state"
5. Add paragraph to introduce full navigation example. "This example uses the basic header and the default, non-collapsible side menu with a sub-menu and page link."


Below feedback proposes shorter sentences and a somewhat more formal tone of voice. Make use of whatever you think is appropriate.

6. Side menu structure can be shortened. 
"The side menu is built up by a wrapper and three main sections. The header, the main content section, and the side menu footer. However the footer sections is only needed if you intend to have a collapsable side menu."
--> 
"The side menu is built up by a wrapper and three sections: mobile header, menu content and menu footer. Footer is used only in collapsible side menu."
7. Information on page links can be more compact. 
"The menu options are to be placed as 'sdds-sidebar-nav__item' within 'sdds-sidebar-nav--main.' 
`<CODE-EXAMPLE>` 
The basic structure of a side menu item is as follows: 
`<CODE-EXAMPLE>`" 
--> 
"Place page links as list items. Basic structure of a side menu with three page links: `
<CODE-EXAMPLE showing combination of both previous code examples>`"

8. Side menu item with sub page options section is a little hard to follow. Suggested changes below:
"To use sup page options on a side menu item firstly add the class 'sdds-sidebar-nav__extended' and the inline svg for the chevron icon to the side menu item container. Then at the end within the side menu item add the sub page container. Note that every sub page container should have a 'sdds-sidebar-nav__item-title' in its first item, this is to display the item title in the sub page options popup in a collapsed state." 
-->
"You can nest lists in side menu to create sub-menus. Style the first item in the sub-menu as 'sdds-sidebar-nav__extended' and add the chevron icon as an inline svg. Repeat the sub-menu name as first list item in the nested list and style as 'sdds-sidebar-nav__item-title'. This displays the sub-menu name above page links when side menu is collapsed and user hovers over it. List subsequent page links as usual."

9. Icon section is a bit "chatty". Suggested changes 
"Icons should be in svg format and have the class 'sdds-sidebar-nav__icon'. Appropriate to use can be found in the icons library. If you opt to not have icons you can simply not add the markup." 
--> 
"Icons are optional for non-collapsible menu but mandatory for the collapsible. Icons should be in inline svg and have the class 'sdds-sidebar-nav__icon'. Appropriate icons are available in the SDDS icon library."

10. "Or if you are looking for a more dynamic solution" --> "If you are looking for a more dynamic solution"

**Comment by areblk** [comment]
I'm having trouble reproducing the code example for side menu item with sub page options. It should be changed to replicate what's in Storybook.

```html
<li class="sdds-sidebar-nav__item sdds-sidebar-nav__extended">
  <a class="sdds-sidebar-nav__item-link">
    <span>Sub-menu</span>
     <svg class="sdds-sidebar-nav__chevron" width="12" height="7" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path></svg>
    <!-- Sub menu -->
    <ul class="sdds-sidebar-nav-subnav">
      <li class="sdds-sidebar-nav__title"><span>Sub-menu</span></li>
      <li class="sdds-sidebar-nav__item">
        <a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
      </li>
      <li class="sdds-sidebar-nav__item">
        <a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
      </li>
    </ul>
    <!-- Sub menu ends -->
  </a>
</li>
```

When I insert it in inspector tool elements editor, it changes and produces this (I added the subnav-open class for comparison)

```html
<li class="sdds-sidebar-nav__item sdds-sidebar-nav__extended subnav-open">
  <a class="sdds-sidebar-nav__item-link">
    <span>Sub-menu</span>
     <svg class="sdds-sidebar-nav__chevron" width="12" height="7" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path></svg>
    <!-- Sub menu -->
    </a><ul class="sdds-sidebar-nav-subnav"><a class="sdds-sidebar-nav__item-link">
      <li class="sdds-sidebar-nav__title"><span>Sub-menu</span></li>
      </a><li class="sdds-sidebar-nav__item"><a class="sdds-sidebar-nav__item-link">
        </a><a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
      </li>
      <li class="sdds-sidebar-nav__item">
        <a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
      </li>
    </ul>
    <!-- Sub menu ends -->
  
</li>
```

image

I think this is because of:
- no end-tag for the initial <a>-element introducing a bug (at least in Edge)
- Wrong class-values for sub menu page link-items
- Wrong class values for sub-menu title item

**Comment by areblk** [comment]
Code example of collapsible bottom menu looks a bit different from Storybook ( no 'sdds-sidebar-toggle' class on icon)

```html
<ul class="sdds-sidebar-nav sdds-sidebar-nav--bottom">
  <li class="sdds-sidebar-nav__item">
    <a class="sdds-sidebar-nav__item-link">
      <svg class="sdds-sidebar-nav__icon" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.046 5.685a1 1 0 0 0-1.414-1.415l-9.9 9.9a2.6 2.6 0 0 0 0 3.678l9.9 9.9a1 1 0 1 0 1.415-1.415L4.722 17.01h24.306a1 1 0 0 0 0-2H4.722l9.325-9.324Z" fill="currentColor"/></svg>
      <span>Collapse</span>
    </a>
  </li>
</ul>
```

Storybook example https://pr-310.d1k9aiuqofg0s0.amplifyapp.com/?path=/story/patterns-sidemenu--collapsible
```html
<ul class="sdds-sidebar-nav sdds-sidebar-nav--bottom icons-enabled">
          <li class="sdds-sidebar-nav__item">
            <a class="sdds-sidebar-toggle sdds-sidebar-nav__item-link" href="#">
              <svg class="sdds-sidebar-nav__icon" fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32">
                <path fill-rule="evenodd" clip-rule="evenodd" d="M14.046 5.685a1 1 0 0 0-1.414-1.415l-9.9 9.9a2.6 2.6 0 0 0 0 3.678l9.9 9.9a1 1 0 1 0 1.415-1.415L4.722 17.01h24.306a1 1 0 0 0 0-2H4.722l9.325-9.324Z" fill="currentColor"></path>
              </svg>
              <span>Collapse</span>
            </a>
          </li>
        </ul>
```

**Comment by areblk** [comment]
Code example for Full page example (basic) looks a bit broken when i try to reproduce it. I'll investigate further after lunch @areblk

**Comment by areblk** [comment]
@victorscania I did all of the changes myself, but I can't wrap my head around how to fix and make working example of a full page. I'll post the original code here and drop it in the documentation for now

```html
  <!-- Header -->
  <nav class='sdds-nav'>
      <div class='sdds-nav__left'>
        <button class='sdds-nav__mob-menu-btn'>
          <svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 32 32"><path fill-rule="evenodd" clip-rule="evenodd" d="M3.97 6.998a1 1 0 0 1 1-1h22.05a1 1 0 0 1 0 2H4.97a1 1 0 0 1-1-1ZM3.97 15.982a1 1 0 0 1 1-1h22.05a1 1 0 0 1 0 2H4.97a1 1 0 0 1-1-1ZM3.97 24.966a1 1 0 0 1 1-1h22.05a1 1 0 0 1 0 2H4.97a1 1 0 0 1-1-1Z" fill="currentColor"/></svg>
        </button>
        <div class='sdds-nav__app-name'>My application</div>
      </div>
      <div class='sdds-nav__right'>
        <a class='sdds-nav__item sdds-nav__app-logo' href='#'></a>
      </div>
  </nav>

  <div class="sdds-push">
    <!-- Side menu -->
    <div class="sdds-sidebar side-menu">
      <!-- Side menu header -->
      <div class="sdds-sidebar-mheader">
        <a href="#" class="sdds-sidebar-mheader__close">
          <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M3.40338 2.34308C3.11048 2.05019 2.63561 2.05019 2.34272 2.34308C2.04982 2.63598 2.04982 3.11085 2.34272 3.40374L6.93897 8L2.34283 12.5961C2.04994 12.889 2.04994 13.3639 2.34283 13.6568C2.63572 13.9497 3.1106 13.9497 3.40349 13.6568L7.99963 9.06066L12.5958 13.6568C12.8887 13.9497 13.3635 13.9497 13.6564 13.6568C13.9493 13.3639 13.9493 12.889 13.6564 12.5961L9.06029 8L13.6565 3.40376C13.9494 3.11086 13.9494 2.63599 13.6565 2.3431C13.3636 2.0502 12.8888 2.0502 12.5959 2.3431L7.99963 6.93934L3.40338 2.34308Z" fill="#171719"/></svg>
        </a>
      </div>
      <!-- Side menu main -->
      <ul class="sdds-sidebar-nav sdds-sidebar-nav--main">
        <!-- Side menu main - Page link -->
        <li class="sdds-sidebar-nav__item">
          <a class="sdds-sidebar-nav__item-link" href="#">
            <span>Page link</span>
          </a>
        </li>
        <!-- Side menu main - Sub menu -->
        <li class="sdds-sidebar-nav__item sdds-sidebar-nav__extended">
          <a class="sdds-sidebar-nav__item-link" href="#">
            <span>Sub-menu</span>
            <svg class="sdds-sidebar-nav__chevron" width="12" height="7" viewBox="0 0 12 7" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M1 1L6 6L11 1" stroke="currentColor" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path></svg>
          </a>
          <ul class="sdds-sidebar-nav-subnav">
            <li class="sdds-sidebar-nav-subnav__item">
              <span class="sdds-sidebar-nav__item-title">Sub-menu</span>
            </li>
            <li class="sdds-sidebar-nav-subnav__item">
              <a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
            </li>
            <li class="sdds-sidebar-nav-subnav__item">
              <a class="sdds-sidebar-nav__item-link" href="#"><span>Sub Page name</span></a>
            </li>
          </ul>
        </li>
        <!-- Side menu main - Page link -->
        <li class="sdds-sidebar-nav__item">
          <a class="sdds-sidebar-nav__item-link" href="#">
            <span>Page link</span>
          </a>
        </li>
      </ul>
    </div>
    <!-- Page Content -->
    <div class="sdds-container">
		  Page content goes here...
	  </div>
  </div>

</body>

**Review by theJohnnyMe — APPROVED** [review]
Looks good to me.
